probe_scraper/runner.py

```diff
@@ -130,58 +130,54 @@ def load_moz_central_probes(cache_dir, out_dir):
     write_moz_central_probe_data(probes_by_channel, revisions, out_dir)
 
 
-def check_git_probe_structure(data):
+def check_glean_metric_structure(data):
     schema = Schema({
         str: {
-            And(str, lambda x: len(x) == 40): {
-                Optional("histogram"): [And(str, lambda x: os.path.exists(x))],
-                Optional("event"): [And(str, lambda x: os.path.exists(x))],
-                Optional("scalar"): [And(str, lambda x: os.path.exists(x))]
-            }
+            And(str, lambda x: len(x) == 40): [And(str, lambda x: os.path.exists(x))]
         }
     })
 
     schema.validate(data)
 
 
-def load_git_probes(cache_dir, out_dir, repositories_file, dry_run):
+def load_glean_metrics(cache_dir, out_dir, repositories_file, dry_run):
     repositories = RepositoriesParser().parse(repositories_file)
-    commit_timestamps, repos_probes_data, emails = git_scraper.scrape(cache_dir, repositories)
+    commit_timestamps, repos_metrics_data, emails = git_scraper.scrape(cache_dir, repositories)
 
-    check_git_probe_structure(repos_probes_data)
+    check_glean_metric_structure(repos_metrics_data)
 
-    # Parse probe data from files into the form:
-    # <repo_name> -> {
-    #   <commit-hash> -> {
-    #     "histogram": {
-    #       <histogram_name>: {
-    #         ...
-    #       },
-    #       ...
-    #     },
-    #     "scalar": {
+    # Parse metric data from files into the form:
+    # <repo_name>:  {
+    #   <commit-hash>:  {
+    #     <metric-name>: {
     #       ...
     #     },
     #   },
     #   ...
     # }
-    probes = defaultdict(lambda: defaultdict(lambda: defaultdict(int)))
-    for repo_name, commits in repos_probes_data.items():
-        for commit_hash, probe_types in commits.items():
-            for probe_type, paths in probe_types.items():
-                try:
-                    results = PARSERS[probe_type].parse(paths)
-                    probes[repo_name][commit_hash][probe_type] = results
-                except Exception:
-                    msg = "Improper file in {}\n{}".format(', '.join(paths), traceback.format_exc())
+    metrics = defaultdict(lambda: defaultdict(lambda: defaultdict(int)))
+    for repo_name, commits in repos_metrics_data.items():
+        for commit_hash, paths in commits.items():
+            try:
+                results, errs = GLEAN_PARSER.parse(paths)
+                metrics[repo_name][commit_hash] = results
+            except Exception:
+                msg = "Improper file in {}\n{}".format(', '.join(paths), traceback.format_exc())
```

Comment:
It may be worth considering a bit more deeply what to do with emails (if anything). In general I'd expect any issues with a `metrics.yaml` file to arise during build, because we're all using the `glean_parser`; in that case, we can perhaps just fail hard (for a specific repo).

Reply:
Since we're talking about GH repos, I wonder if there's an API to file issues :) (definitely not blocking, just a loud thought)